dashboard/src/components/settings/authentication/BlockedEmailSettings/BlockedEmailSettings.tsx

```diff
@@ -79,31 +70,53 @@ export default function BlockedEmailSettings() {
   const handleAllowedEmailDomainsChange = async (
     values: BlockedEmailFormValues,
   ) => {
-    const updateAppMutation = updateApp({
+    const updateConfigPromise = updateConfig({
       variables: {
-        id: currentApplication.id,
-        app: {
-          authAccessControlBlockedEmails: values.enabled
-            ? values.authAccessControlBlockedEmails
-            : '',
-          authAccessControlBlockedEmailDomains: values.enabled
-            ? values.authAccessControlBlockedEmailDomains
-            : '',
+        appId: currentApplication.id,
+        config: {
+          auth: {
+            user: {
+              email: {
+                allowed: email.allowed,
+                blocked:
+                  values.enabled && values.blockedEmails
+                    ? values.blockedEmails
+                        .split(',')
+                        .map((blockedEmail) => blockedEmail.trim())
+                    : [],
+              },
+              emailDomains: {
+                allowed: emailDomains.allowed,
+                blocked:
+                  values.enabled && values.blockedEmailDomains
+                    ? values.blockedEmailDomains
+                        .split(',')
+                        .map((blockedEmailDomain) => blockedEmailDomain.trim())
```

Comment:
I would probably do the split/map/trip operation above and also include removing duplicates.

```js
[...new Set(array)];
```